source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -511,8 +511,8 @@ void DnsCacheImpl::addCacheEntry(
     value = absl::StrCat(address->asString(), "|", ttl.count(), "|", seconds_since_epoch);
   } else {
     for (auto& addr : address_list) {
-      value += absl::StrCat((value.empty() ? "" : "\n"), addr->asString(), "|", ttl.count(), "|",
-                            seconds_since_epoch);
+      absl::StrAppend(&value, (value.empty() ? "" : "\n"), addr->asString(), "|", ttl.count(), "|",
```

Comment:
ISTM that in the end we're just doing a StrJoin with newline as the seperator. Can we do that instead then?

Reply:
StrJoin is more for a vector or another objects `Ranges are specified by passing a container with std::begin() and std::end() iterators, container-specific begin() and end()`, we want to avoid temporary strings, and iiuc using append here is the most efficient way.

Reply:
see https://abseil.io/docs/cpp/guides/strings#abslstrjoin-for-joining-elements-within-a-string.

We could achieve this using a custom formatter lambda and get rid of this raw loop assigning the output string to value.

Reply:
Thanks, done.